src/analysis/try_from_glib.rs

```diff
@@ -0,0 +1,101 @@
+use std::{borrow::Cow, sync::Arc};
+
+use crate::{
+    analysis::conversion_type::ConversionType,
+    config,
+    library::{self, Infallible, Mandatory},
+    Env,
+};
+
+#[derive(Clone, Debug)]
+pub enum TryFromGlib {
+    Default,
+    NotImplemented,
+    Option,
+    OptionMandatory,
+    Result {
+        ok_type: Arc<str>,
```

Comment:
Why `Arc` and not `Rc`?

Reply:
I guess `gir` must be using some non-`Sync` mechanisms somewhere.

`rustc` forced me to use `Arc` instead of `Rc`. Didn't want to investigate why that was needed as it was not the point of this PR and I had already been through a lot to get there.

Reply:
The reason is simply because we store values of these things in `static`s:

```
error[E0277]: `Rc<str>` cannot be sent between threads safely
  --> src/codegen/sys/functions.rs:15:1
   |
15 | static DEFAULT_OBJ: Lazy<GObject> = Lazy::new(Default::default);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<str>` cannot be sent between threads safely
   |
   = help: within `GObject`, the trait `std::marker::Send` is not implemented for `Rc<str>`
   = note: required because it appears within the type `ConversionType`
   = note: required because it appears within the type `std::option::Option<ConversionType>`
   = note: required because it appears within the type `GObject`
   = note: required because of the requirements on the impl of `Sync` for `once_cell::imp::OnceCell<GObject>`
   = note: required because it appears within the type `once_cell::sync::OnceCell<GObject>`
   = note: required because of the requirements on the impl of `Sync` for `once_cell::sync::Lazy<GObject>`
   = note: shared static variables must have a type that implements `Sync`
```

If we don't want `Arc` then we'd have to replace this on all usage sites with `GObject::default()`. If that's better or not, I don't know, but it's certainly not the point of this PR.

Reply:
Yes, that's the one. Sorry for not trying that again to illustrate.
It's probably just a matter of using [`unsync::Lazy`](https://docs.rs/once_cell/1.7.2/once_cell/unsync/struct.Lazy.html).

I can try this and update if the PR doesn't stall too long after that :)